### PR TITLE
Add native trace API

### PR DESCRIPTION
**Pull request by JasperDeSutter**
To test tracing functionality, see: https://developer.android.com/topic/performance/tracing/on-device

### CODE REVIEW

File: ndk/src/trace.rs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+#![cfg(feature = "trace")]
+use std::ffi::{CString, NulError};
+
+pub fn is_trace_enabled() -> bool {
+    unsafe { ffi::ATrace_isEnabled() }
+}
+
+#[derive(Debug)]
+pub struct Section {
+    _private: *mut (),
```
Comment (msiglreith) [inline ndk/src/trace.rs:13]: minor: a comment should be added why this field is required. I would also suggest to rather use `PhantomMarker` if it's soley to enforce !Send+!Sync

Comment (JasperDeSutter) [reply ndk/src/trace.rs]: Indeed, didn't have to use a pointer here. Thanks for the suggestion!

File: ndk/src/trace.rs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    _pd: PhantomData<&'static ()>,
+}
+
+#[cfg(feature = "api-level-29")]
+impl AsyncSection {
+    pub fn new(name: &str, cookie: Cookie) -> Result<Self, NulError> {
+        let section_name = CString::new(name)?;
+        unsafe { ffi::ATrace_beginAsyncSection(section_name.as_ptr(), cookie.0) };
+
+        Ok(AsyncSection {
```
Comment (francesca64) [inline ndk/src/trace.rs:57]:
```suggestion
        Ok(Self {
```

File: ndk/src/trace.rs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    // Section is !Sync and !Send
+    _pd: PhantomData<*mut ()>,
+}
+
+impl Section {
+    pub fn new(name: &str) -> Result<Self, NulError> {
+        let section_name = CString::new(name)?;
+        unsafe { ffi::ATrace_beginSection(section_name.as_ptr()) };
+
+        Ok(Section { _pd: PhantomData })
```
Comment (francesca64) [inline ndk/src/trace.rs:23]:
```suggestion
        Ok(Self { _pd: PhantomData })
```

**Pull request merged**